Content.Server/GameObjects/Components/Movement/ClimbableComponent.cs

```diff
@@ -0,0 +1,223 @@
+        {
+            if (user.TryGetComponent(out ICollidableComponent body) && body.PhysicsShapes.Count >= 1)
+            {
+                var direction = (Owner.Transform.WorldPosition - entityToMove.Transform.WorldPosition).Normalized;
+                var endPoint = Owner.Transform.WorldPosition;
+
+                var climbMode = entityToMove.GetComponent<ClimbModeComponent>();
+                climbMode.SetClimbing(true);
+
+                if (MathF.Abs(direction.X) < 0.6f) // user climbed mostly vertically so lets make it a clean straight line
+                {
+                    endPoint = new Robust.Shared.Maths.Vector2(entityToMove.Transform.WorldPosition.X, endPoint.Y);
+                }
+                else if (MathF.Abs(direction.Y) < 0.6f) // user climbed mostly horizontally so lets make it a clean straight line
+                {
+                    endPoint = new Robust.Shared.Maths.Vector2(endPoint.X, entityToMove.Transform.WorldPosition.Y);
+                }
+
+                climbMode.TryMoveTo(entityToMove.Transform.WorldPosition, endPoint);
+                // we may potentially need additional logic since we're forcing a player onto a climbable
+                // there's also the cases where the user might collide with the person they are forcing onto the climbable that i haven't accounted for
+
+                PopupMessageOtherClientsInRange(user, Loc.GetString("{0:them} forces {1:them} onto {2:theName}!", user, entityToMove, Owner), 15);
+                _notifyManager.PopupMessage(user, user, Loc.GetString("You force {0:them} onto {1:theName}!", entityToMove, Owner));
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool TryClimb(IEntity user)
+        {
+            if (user.TryGetComponent(out ICollidableComponent body) && body.PhysicsShapes.Count >= 1)
+            {
+                var direction = (Owner.Transform.WorldPosition - user.Transform.WorldPosition).Normalized;
+                var endPoint = Owner.Transform.WorldPosition;
+
+                var climbMode = user.GetComponent<ClimbModeComponent>();
+                climbMode.SetClimbing(true);
+
+                if (MathF.Abs(direction.X) < 0.6f) // user climbed mostly vertically so lets make it a clean straight line
+                {
+                    endPoint = new Robust.Shared.Maths.Vector2(user.Transform.WorldPosition.X, endPoint.Y);
+                }
+                else if (MathF.Abs(direction.Y) < 0.6f) // user climbed mostly horizontally so lets make it a clean straight line
+                {
+                    endPoint = new Robust.Shared.Maths.Vector2(endPoint.X, user.Transform.WorldPosition.Y);
+                }
+
+                climbMode.TryMoveTo(user.Transform.WorldPosition, endPoint);
+
+                PopupMessageOtherClientsInRange(user, Loc.GetString("{0:them} jumps onto the {1}!", user, Owner), 15);
```

Comment:
Is "them" what you wanted to use here? Also sucks because "He jumps" "She jumps" "They jump" the verb changes depending on the subject. This isn't your fault though, as I don't think we have a macro for this yet.

Reply:
Also you put `the` when you should probably use `{1:theName}` again.

Reply:
This still needs to be resolved with `"{0:them} jumps onto {1:theName}!"`